### ISSUE TITLE
Allow user to set name tag value when creating workers using launch templates

### ISSUE THREAD

**Issue by aydasraf** [terraform-aws-modules/terraform-aws-eks#1688]
## Is your request related to a new offering from AWS?
No

## Is your request related to a problem? Please describe.
Yes, I am creating an EKS cluster using version `17.23.0` using `worker_groups_launch_template`, I am trying to set name tags for the instances to our naming standard to allow domain join for those nodes, by setting the name tag in the code like below: 
```hcl
tags  = [
          {
            key                 = "Name"
            value               = "xxx-yyy-eks"
            propagate_at_launch = true
          },
          {
            key                 = "ADJoin"
            value               = true
            propagate_at_launch = true
          }
        ]
```

but sadly the module overrides the works by using force `cluster_name - work_group_name-eks_sg`, code here:  https://github.com/terraform-aws-modules/terraform-aws-eks/blob/4e54eaac16f8e1df0c71abb56eca71633a8a0b41/workers_launch_template.tf#L550-L606  which causes the domain join to fail and the nodes to be outside of our org naming convention causing a huge mess. 

## Describe the solution you'd like.
if the `name` tag is set in the tags, respect that and use these values, if not build the value as previously done.
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/4e54eaac16f8e1df0c71abb56eca71633a8a0b41/workers_launch_template.tf#L550-L606

## Describe alternatives you've considered.
Manually overriding the `workers_launch_template.tf` to fulfill this, is not convenient and caused a couple of incidents! 

## Additional context
N/A

**Comment by aydasraf** [comment]
Hello Again, Any Updates on this issue?

**Comment by antonbabenko** [comment]
This issue has been resolved in version 18.0.0 :tada:
